chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -279,383 +313,398 @@ public static double getDouble(final IoBuffer readBuffer) {
+            for (int i = 0; i < nElements; i++) {
+                retCollection.add(serialiser.getReturnObjectFunction().apply(this, null, null));
+            }
 
-    public static String getString(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getString();
+            return retCollection;
         }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
 
-    public static String[] getStringArray(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getStringArray();
+        // read primitive or String value vector
+        final E[] values = getGenericArrayAsBoxedPrimitive(valueDataType);
+        if (nElements != values.length) {
+            throw new IllegalStateException(PROTOCOL_MISMATCH_N_ELEMENTS_HEADER + nElements + VS_ARRAY + values.length);
         }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
-
-    public static FieldHeader parseIoStream(final IoBuffer buffer) {
-        final FieldHeader fieldRoot = new FieldHeader("ROOT", DataType.OTHER, new int[] { 0 }, buffer.position(), 100);
-        final HeaderInfo headerRoot = checkHeaderInfo(buffer);
-        headerRoot.setParent(fieldRoot);
-        fieldRoot.getChildren().add(headerRoot);
+        retCollection.addAll(Arrays.asList(values));
 
-        parseIoStream(buffer, headerRoot, 0);
-        return fieldRoot;
+        return retCollection;
     }
 
-    protected static void parseIoStream(final IoBuffer buffer, final FieldHeader fieldRoot, final int recursionDepth) {
-        FieldHeader fieldHeader;
-        while ((fieldHeader = BinarySerialiser.getFieldHeader(buffer)) != null) {
-            final long bytesToSkip = fieldHeader.getExpectedNumberOfDataBytes();
-            final long skipPosition = buffer.position() + bytesToSkip;
-            fieldHeader.setParent(fieldRoot);
-
-            // reached end of (sub-)class - check marker value and close nested hierarchy
-            if (fieldHeader.getDataType() == DataType.END_MARKER) {
-                // reached end of (sub-)class - check marker value and close nested hierarchy
-                final byte markerValue = buffer.getByte();
-                final Optional<FieldHeader> superParent = fieldRoot.getParent();
-                if (DataType.END_MARKER.getAsByte() != markerValue) {
-                    throw new IllegalStateException("reached end marker, mismatched value '" + markerValue + "' vs. should '" + DataType.END_MARKER.getAsByte() + "'");
-                }
-
-                if (superParent.isEmpty()) {
-                    fieldRoot.getChildren().add(fieldHeader);
-                } else {
-                    superParent.get().getChildren().add(fieldHeader);
-                }
-                break;
+    @Override
+    public <E> Set<E> getSet(final Set<E> collection, BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
```

Comment:
Codacy found an issue: [The method 'getSet(Set, BiFunction)' has an NPath complexity of 204, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)